File: js/task2/S2_FOR_02.js
// S2_FOR_02 — Find first even

function findFirstEven(nums) {
  for (const n of nums) {
    if (n % 2 === 0) {
      return n;
    }
  }
  return null;
}

// Tests
console.log(findFirstEven([1, 3, 5, 6, 8])); // 6
console.log(findFirstEven([1, 3, 5]));       // null
console.log(findFirstEven([2, 4, 6]));       // 2

File: python/task2b/S2_FOR_02.py
# S2_FOR_02 — Find first even using a loop

def find_first_even(nums):
    for n in nums:
        if n % 2 == 0:
            return n
    return None


def main():
    tests = [
        [1, 3, 5, 6, 7],
        [1, 3, 5],
        [],
        [2, 4, 6],
    ]

    for t in tests:
        print(f"in={t} -> first_even={find_first_even(t)}")


if __name__ == "__main__":
    main()
